pkg/controller/persistentvolume/persistentvolume_provisioner_controller.go

```diff
@@ -345,6 +347,11 @@ func newProvisioner(plugin volume.ProvisionableVolumePlugin, claim *api.Persiste
 		AccessModes:                   claim.Spec.AccessModes,
 		PersistentVolumeReclaimPolicy: api.PersistentVolumeReclaimDelete,
 		CloudTags:                     &tags,
+		ClusterName:                   controller.clusterName,
+	}
+
+	if pv != nil {
+		volumeOptions.PVName = pv.Name
```

Comment:
Got it, thanks Mark! Yes, a comment would help.

So one more question: why set the final name in such a roundabout fashion? i.e. `NewPersistentVolumeTemplate()` will set it to `pv-aws-UID` then `CreateVolume(...)` will set it to `clusterName-dynamic-pv-aws-UID`. Why not just set it to the final name directly in `NewPersistentVolumeTemplate()`?